subsys/net/lib/download_client/src/download_client.c

```diff
@@ -14,6 +14,11 @@
 #include <net/download_client.h>
 #include <logging/log.h>
 
+#if ! __GNU_VISIBLE
+/* strcasestr exists in libc and can be linked, but __GNU_VISIBLE must be 1 to be available in string.h */
+char	*strcasestr (const char *, const char *);
```

Comment:
`strcasestr` is a non-standard function and is not available in Zephyr's minimal libc implementation (Zephyr default). In result, netither of our FOTA samples builds with this patch.

I think therefore that we should approach the problem in some other way. Normalize the header case before parsing it perhaps?